server/src/main/resources/com/paypal/selion/js/listAllNodes.js

```diff
@@ -0,0 +1,209 @@
+        proxyInfoLeftColumnDiv = $('<div>').attr('class', 'left-cell'),
+        proxyInfoRightColumnDiv = $('<div>').attr('class', 'right-cell'),
+        // strips the package from the proxy class name and returns the result
+        proxyClass = (function () {
+          var p = node.configuration.proxy,
+            indexOf = p.lastIndexOf('.');
+          if (indexOf === -1) {
+            return p;
+          }
+          return p.substr(indexOf + 1, p.length);
+        }());
+
+      // add the proxy header row
+      function createProxyHeaderDiv() {
+        var div = $('<div>'),
+          // sets the header css class based on proxy status
+          headerCssClass = (function () {
+            var cssClass = 'header';
+            if (node.status === 'offline') {
+              cssClass += ' offline';
+            }
+            if (node.isShuttingDown) {
+              cssClass += ' shuttingdown';
+            }
+            return cssClass;
+          }());
+
+        // adds the node version and os, if available
+        function createVersionAndOSSpan() {
+          var span = $('<span>').attr('class', 'version-info');
+          if (node.status === 'online') {
+            $(span).text('v' + node.version);
+          }
+          if (node.os !== 'not available') {
+            $(span).text($(span).text() + ' on ' + node.os);
+          }
+          return span;
+        }
+
+        // add the remote host / slot id info
+        function createRemoteHostSpan() {
+          var span = $('<span>');
+          if (node.logsLocation !== 'not available') {
+            $('<a>')
+              .attr('target', '_blank')
+              .attr('href', node.logsLocation)
+              .text(node.configuration.remoteHost)
+              .appendTo(span);
+          } else {
+            $(span).text(node.configuration.remoteHost);
+          }
+          return span;
+        }
+
+        return $(div)
+          .attr('class', headerCssClass)
+          .append(createRemoteHostSpan())
+          .append(createVersionAndOSSpan());
+      }
+
+      // add the slot info
+      function addSlotInfo(slots) {
+        var slotsDiv = $('<div>')
+          .attr('class', 'icon-row')
+          .attr('data-src', index);
+        if (typeof slots !== 'object') {
+          return;
+        }
+        $.each(slots, function addli(k, v) {
+          $(slotsDiv).append($('<li>').attr('class', 'icon').attr('data-value', k));
+        });
+        return slotsDiv;
+      }
+
+      // add a <li> item to display
+      function createInfoLi(label, item, valueType, isSelionItem) {
+        // sanitize known values which mean the proxy did not provide any data.
+        function supported(something) {
+          if (proxyClass === 'SeLionRemoteProxy') {
+            return something;
+          }
+          try {
+            if ((something === undefined) || (something === null) || (something === -1)) {
```

Comment:
We can update this as if (!something || (something === -1)). if (something) will validate false, 0, "", NaN, null, 
undefined.